src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,5 +60,42 @@ public boolean equals(Object other) {
     public int hashCode() {
         return fullName.hashCode();
     }
+    
+    /**
+     * Returns true if the other name is very similar to this name.
+     * Two names are considered similar if they are: (ignoring case)
+     * equal, in different order, or if one name is a subset of the other name.
+     * Note that the subset definition we use includes the former two cases as well.
+     */
+     public boolean isSimilar(Name other) { 
+         if (other == null) {
+             return false;
+         }
+         String thisNameIgnoreCase = this.fullName.toLowerCase();
+         String otherNameIgnoreCase = other.fullName.toLowerCase();
+
+         return isOneNameSubsetOfTheOther(thisNameIgnoreCase, otherNameIgnoreCase);
```

Comment:
consider renaming this method? it does multiple other things, though the overall method is check one subset of other, to check if the names are similar!